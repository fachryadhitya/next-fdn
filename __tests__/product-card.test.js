import { render } from '@testing-library/react';
import ProductCard from '../components/product-card';
import '@testing-library/jest-dom'

describe('Product Card Component', () => {
  it('Render successfully', () => {
    const { container } = render(<ProductCard />);
    expect(container).toMatchSnapshot();
  });

  it('Render component with custom props', () => {
    const { getByText, getByAltText } = render(<ProductCard matchSkin withProfileImage />);
    expect(getByText('match skin type')).toBeInTheDocument();
    expect(getByAltText('editor profile')).toBeInTheDocument();
  });
});
