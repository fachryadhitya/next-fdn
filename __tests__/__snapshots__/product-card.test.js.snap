// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Component Render successfully 1`] = `
<div>
  <div
    class="w-full md:w-1/3 lg:w-48% xl:w-18% border-2 rounded-lg relative my-6 undefined"
  >
    <div
      class="bg-white p-4 rounded-lg "
    >
      <img
        alt="content"
        class="h-40 rounded w-full object-contain object-center mb-6 max-w-full"
      />
      <span
        class="flex items-center"
      >
        <span
          class="mr-2"
        />
        <span
          class="ml-2"
        >
          (2)
        </span>
      </span>
      <h2
        class="text-lg text-gray-900 font-medium title-font my-2 uppercase"
      />
      <p
        class="leading-relaxed text-base line-clamp-2"
      />
      <p
        class="text-base text-neutral-400"
      >
         Rosy Beige
      </p>
    </div>
  </div>
</div>
`;
